src/opower/opower.py

```diff
@@ -529,9 +540,12 @@ def _get_headers(self, customer_uuid: Optional[str] = None) -> dict[str, str]:
         opower_selected_entities = []
         if self.utility.is_dss() and self.user_accounts:
             # Required for dss endpoints
-            opower_selected_entities.append(
-                f'urn:session:account:{self.user_accounts[0]["accountId"]}'
-            )
+            account_id = self._get_account_id()
+            if account_id:
+                opower_selected_entities.append(f"urn:session:account:{account_id}")
+            else:
+                raise ValueError("Can't find any accounts with premises")

```

Comment:
Could you instead fallback to the old behavior?

Reply:
Added an old code but with comment about this case